src/modules/net/BOAClient.ts

```diff
@@ -192,6 +192,22 @@ export class BOAClient
             message: "The pre-image is valid."
         };
     }
+
+    /**
+     * TODO As this might get influenced by future changes
+     * Shell function to convert from time to height
+     * @param when {Date} Unix epoch time
+     * @returns height {number | undefined} (or expected height) of the designated time
+     */
+    public getHeightAt(when: Date): number | undefined
+    {
+        const baseDate: Date = new Date(Date.UTC(2020,0,1,0,0,0));
+        if (when.getTime() < baseDate.getTime())
+            return undefined;
```

Comment:
```suggestion
     * @returns height {number} (or expected height) of the designated time
     */
    public getHeightAt(when: Date): number
    {
        const baseDate: Date = new Date(Date.UTC(2020,0,1,0,0,0));
        if (when.getTime() < baseDate.getTime())
            return Number.NaN;
```
 I think `Number.NaN` is better than `undefine` because `undefine` means optional return types.

Reply:
https://github.com/bpfkorea/boa-sdk-ts/blob/85166ac32dc3f3687e2835e1bbeeb50cd909e70c/src/modules/net/BOAClient.ts#L49-L54

https://github.com/bpfkorea/boa-sdk-ts/blob/85166ac32dc3f3687e2835e1bbeeb50cd909e70c/src/modules/net/BOAClient.ts#L99-L104

We already have common use of the number | undefended specification for height.

Reply:
This is a parameter. The current one is the return value.
I think they are different.

Reply:
Could you tell me the advantages of returning `Number.NaN`?

Reply:
An error occurs in this case.
``` TypeScript
        let date = new Date(Date.UTC(2019,3,29,0,0,0));
        let height:Number = boa_client.getHeightAt(date);
        console.log(height);
```